src/ripple/conditions/impl/fulfillment.cpp

```diff
@@ -0,0 +1,211 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2016 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ripple/conditions/condition.h>
+#include <ripple/conditions/fulfillment.h>
+#include <ripple/conditions/hashlock.h>
+#include <ripple/conditions/prefix_sha256.h>
+#include <ripple/conditions/ed25519.h>
+#include <ripple/conditions/rsa_sha256.h>
+#include <ripple/conditions/impl/utils.h>
+#include <boost/regex.hpp>
+#include <boost/optional.hpp>
+#include <type_traits>
+#include <vector>
+
+namespace ripple {
+namespace cryptoconditions {
+
+bool
+fulfills (
+    fulfillment_t const& f,
+    condition_t const& c)
+{
+    // Fast check: the fulfillment's type must match the
+    // conditions's type:
+    if (f.type() != c.type)
+        return false;
+
+    return f.condition() == c;
+}
+
+bool
+validate (
+    fulfillment_t const& f,
+    condition_t const& c,
+    Slice const& m)
```

Comment:
It's obvious what the `f` and `c` parameters are. `m` is less obvious. Rename to `message`?